nevergrad/functions/games/battleship.py

```diff
@@ -0,0 +1,337 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+#
+# We have two versions of the game:
+# version 1: oteytaud's version, simple.
+# version 2: version as in https://stackoverflow.com/questions/1631414/what-is-the-best-battleship-ai
+#        also consistent with https://github.com/Dbz/Battleship/blob/master/BattleShip.py
+#
+#
+# Opponents:
+#   undocumented opponents at https://stackoverflow.com/questions/1631414/what-is-the-best-battleship-ai
+#   shooting policy in https://github.com/Dbz/Battleship/blob/master/BattleShip.py
+#   full policies in https://github.com/Zulban/battleship-ai
+#         random: uniform random shooting (even redundant!), one ship per row (max) with random rows, all horizontal
+#         ocd: shoot the first place which was not shot; fixed positioning:
+#           5: horizontal at (2,2)
+#           4: horizontal at (3,2)
+#           3: horizontal at (4,2)
+#           3: horizontal at (5,2)
+#           2: Vertical at   (6,7)
+#   shooting policy in http://christopherstoll.org/2012/06/battleship-ai-algorithm-using-dynamic.html
+
+
+import random
+from collections import defaultdict
+
+
+class battleship(object):
```

Comment:
classes should be named with capital letters. Also, in python3 you should remove `(object)` 